panic: assignment to entry in nil map

This happened on two machines running nomad in server mode.

I know I'm running an older version, but submitting this bug in case it is still an issue in the latest release.



### Nomad version
Output from `nomad version`
Nomad v0.5.2

### Operating system and Environment details
Linux ip-10-201-0-8 4.4.0-66-generic #87-Ubuntu SMP Fri Mar 3 15:29:05 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

### Issue
```
panic: assignment to entry in nil map

goroutine 11240414 [running]:
panic(0xfc0740, 0xc71652ac10)
        /opt/go/src/runtime/panic.go:500 +0x1a1
github.com/hashicorp/nomad/nomad.evaluatePlan.func1(0xc70eec3aa0, 0x24, 0xc72274fb01, 0x0, 0x0, 0x1)
        /opt/gopath/src/github.com/hashicorp/nomad/nomad/plan_apply.go:252 +0x2c2
github.com/hashicorp/nomad/nomad.evaluatePlan(0xc763b18240, 0xc6b94ad260, 0xc701eba500, 0x0, 0x0, 0x0)
        /opt/gopath/src/github.com/hashicorp/nomad/nomad/plan_apply.go:276 +0xad4
github.com/hashicorp/nomad/nomad.(*Server).planApply(0xc4201fb1e0)
        /opt/gopath/src/github.com/hashicorp/nomad/nomad/plan_apply.go:82 +0x171
created by github.com/hashicorp/nomad/nomad.(*Server).establishLeadership
        /opt/gopath/src/github.com/hashicorp/nomad/nomad/leader.go:123 +0xde
    Loaded configuration from /etc/nomad/config.hcl
```


### Reproduction steps
None

### Nomad Server logs (if appropriate)

[logfile.txt](https://github.com/hashicorp/nomad/files/904106/logfile.txt)

--- a/nomad/plan_apply.go
+++ b/nomad/plan_apply.go
@@ -263,6 +263,7 @@ func evaluatePlan(pool *EvaluatePool, snap *state.StateSnapshot, plan *structs.P
 
 	// Evalute each node in the plan, handling results as they are ready to
 	// avoid blocking.
+OUTER:
 	for len(nodeIDList) > 0 {
 		nodeID := nodeIDList[0]
 		select {
@@ -276,7 +277,7 @@ func evaluatePlan(pool *EvaluatePool, snap *state.StateSnapshot, plan *structs.P
 			// which may save time processing additional entries.
 			if cancel := handleResult(r.nodeID, r.fit, r.err); cancel {
 				didCancel = true
-				break
+				break OUTER
 			}
 		}
 	}

